lib/route.js

```diff
@@ -426,9 +427,7 @@ internals.payload = async function (request) {
     catch (err) {
         Bounce.rethrow(err, 'system');
 
-        if (request._isPayloadPending) {
-            await internals.drain(request);
-        }
+        await internals.drain(request);
```

Comment:
That is not a concern from this PR, right?

I tend to agree. The current `drain()` logic requires that the stream is active to work. An already closed stream would stall forever.

I would be wary to rely on `stream.finished()`, as I have seen streams that causes issues with it in the past.

Reply:
Not from this PR no, but removing that conditional made me look further whether it was solid or not, so I hope there's something to consume there, otherwise it's indeed stalled.

Reply:
FYI, draining might not be the best approach, but I retained the current behaviour.

The alternative is to just `.destroy()` the stream, which should work fine, but for HTTP/1 it will interrupt any pipelined requests on the connection. Draining is more graceful, but can needlessly use up bandwidth. It works best for small pending payloads.